drivers/sensor/stts22h/stts22h.c

```diff
@@ -0,0 +1,208 @@
+/* ST Microelectronics STTS22H temperature sensor
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/stts22h.pdf
+ */
+
+#define DT_DRV_COMPAT st_stts22h
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "stts22h.h"
+
+LOG_MODULE_REGISTER(STTS22H, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int stts22h_set_odr_raw(const struct device *dev, stts22h_odr_temp_t odr)
+{
+	const struct stts22h_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	return stts22h_temp_data_rate_set(ctx, odr);
+}
+
+static int stts22h_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct stts22h_data *data = dev->data;
+	const struct stts22h_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	int16_t raw_temp;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
```

Comment:
I think this is only going to compile in if asserts are enabled, you really want it to return error all the time, just

```c
if (chan != SENSOR_CHAN_ALL) {
        return -ENOTSUP;
}
```

Reply:
OK, even if it seems that calling ASSERT is a pretty common way for a sensor fetch.